server/src/server_host.ts

```diff
@@ -0,0 +1,173 @@
+/**
+ * `ServerHost` is a wrapper around `ts.sys` for the Node system. In Node, all
+ * optional methods of `ts.System` are implemented.
```

Comment:
I'm a bit confused by this. How do we know that the `ts.System` passed to the constructor will be the system created from `sys.getNodeSystem`?

Reply:
Sorry for the confusion. This is meant to be used by the new `server.ts`. But you're right, instead of accepting `ts.sys` as an input, it should just import it directly.

If TypeScript detects that it's running on Node, it'd create `sys = getNodeSystem()`.
https://github.com/microsoft/TypeScript/blob/25f609b3a22996ba576ff6d6254567a36cc274b6/src/compiler/sys.ts#L1382-L1386